craft_parts/executor/part_handler.py

```diff
@@ -0,0 +1,384 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2017-2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Definitions and helpers for part handlers."""
+
+import logging
+import os
+import os.path
+import shutil
+from glob import iglob
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Optional
+
+from craft_parts import callbacks, errors, packages, plugins, sources
+from craft_parts.actions import Action, ActionType
+from craft_parts.infos import PartInfo, StepInfo
+from craft_parts.packages import errors as packages_errors
+from craft_parts.parts import Part
+from craft_parts.plugins import Plugin
+from craft_parts.state_manager import states
+from craft_parts.state_manager.states import StepState
+from craft_parts.steps import Step
+from craft_parts.utils import os_utils
+
+from .organize import organize_files
+from .step_handler import StepContents, StepHandler
+
+logger = logging.getLogger(__name__)
+
+
+class PartHandler:
+    """Handle steps for a part using the appropriate plugins.
```

Comment:
Is mentioning "using the appropriate plugins" necessary considering that you pass a Part (which encapsulates the plugin)?

```suggestion
    """Manage the life cycle steps for a Part.
```

I might be missing doc hints for sphinx here